prolif/protein_helper.py

```diff
@@ -0,0 +1,560 @@
+        )
+
+    @staticmethod
+    def n_standard_residue_heavy_atoms(resname: str) -> int:
+        """Count the number of heavy atoms in a standard residue.
+
+        Parameters
+        ----------
+        resname : str
+            The residue name to count the heavy atoms in.
+
+        Returns
+        -------
+        int
+            The number of heavy atoms in the standard residue.
+        """
+        if resname not in STANDARD_AA:
+            raise ValueError(f"Residue {resname} is not a standard residue.")
+        residue_atom_df = STANDARD_AA[resname]["_chem_comp_atom"]
+        residue_atom_df = residue_atom_df[residue_atom_df["alt_atom_id"] != "OXT"]
+
+        return sum(residue_atom_df["type_symbol"] != "H")
+
+    @staticmethod
+    def fix_molecule_bond_orders(
+        residue: Residue, templates: list | None = None
+    ) -> Residue:
+        """Fix the bond orders of a molecule.
+
+        Parameters
+        ----------
+        residue : Residue
+            The residue to fix the bond orders for.
+        templates : str or list of str, optional
+            The templates to use for fixing the bond orders.
+            If `None`, the standard amino acid template is used.
+            Default is `None`. If the residue is not a standard amino acid,
+            the user should provide a custom template.
+
+            Also, note that the order of templates is relevant,
+            as the first template that matches the residue name will be used.
+
+        Note
+        ----
+        Any bonds and chiral designation on the input molecule will be removed
+        at the start of the process.
+        This function is adapted from the pdbinf/_pdbinf.py module's assign_pdb_bonds
+        function, which is used to assign bonds and aromaticity based on
+        the standard amino acid templates.
+        """
+        templates = [STANDARD_AA] if templates is None else [templates, STANDARD_AA]
+        resname = residue.resid.name.upper()
+
+        # strip bonds and chiral tags
+        new_res = strip_bonds(residue)
+
+        # assign properties inside each Residue
+        valence = np.zeros(new_res.GetNumAtoms(), dtype=int)
+
+        for t in templates:
+            # check if resname in template doc
+            if resname not in t:
+                continue
+            t[resname]["name"] = resname  # add name for the reference block
+            new_res, v = assign_intra_props(new_res, t[resname])
+            valence += v
+            break  # avoid double assignment, ordering of templates is relevant
+        else:
+            raise ValueError(f"Failed to find template for residue: '{resname}'")
+
+        return Residue(new_res)
+
+
+# The below code contains functions for fixing molecule bond orders.
+# The idea is based on pdbinf: https://github.com/OpenFreeEnergy/pdbinf/tree/main
+# A user can provide a custom CIF file with the standard amino acid.
+def _block_decompose(data_block: list) -> tuple:
```

Comment:
this function and `cif_parser_lite` should be in a separate `cif` module. Maybe we should have a `io` folder to contain this and the protein helper module